Rendera/include/scene.hpp

```diff
@@ -0,0 +1,171 @@
+#ifndef SCENE_HPP
+#define SCENE_HPP
+
+#include "sphere.hpp"
+#include "intersection.hpp"
+#include "camera.hpp"
+
+class Hitinfo{
+    public:
+        bool ishitting = false;
+        float dist;
+        vec3 intersection_pt;
+        ray normal;
+        ray ray_;
+        color shade = {0,0,0};
+        color shade_new = {0,0,0};
+        color rgb;
+        ray ref_ray;
+};
+
+color lighting(Material mat, ray cam_ray, ray normal, vec3 point, vec3 light_src, color intensity){
+
+    ray incident_ray = {point, point - light_src};
+    ray reflect_ray = reflect(incident_ray, normal);
+
+    color effective_col = mat.Color() * intensity;
+
+    //determine ambient contribution
+    color ambient_col = effective_col * mat.amb();
+
+    
+    //determine diffuse contribution, using Lambert's cosine law
+    color diffuse_col = {0, 0, 0};
+    float diffuse_cosine = dot(-incident_ray.get_direction(), normal.get_direction());
+
+    // float lambert_cosine = dot(-cam_ray.get_direction(), normal.get_direction());
+
+    if (diffuse_cosine >= 0){
+        diffuse_col = diffuse_cosine * effective_col * mat.dif();
+    }
+
+    
+
+    float specular_cosine = dot(-cam_ray.get_direction(), reflect_ray.get_direction());
+    float spec_factor = 0;
+
+    if(specular_cosine >= 0){
+        spec_factor = pow(specular_cosine, mat.shiny());
+    }
+    color specular_col = spec_factor * intensity * mat.spe();
+
+    return ambient_col + diffuse_col + specular_col;
+}
+
+class Scene{
+    public:
+        int n_sphere;
+        int img_width;
+        int img_height;
+        Hitinfo *hitinfo = new Hitinfo[img_height*img_width];
+        sphere *hitable_sphere = new sphere[n_sphere];
+
+
+        vec3 light_src = {-5,3,5};
+        color light_col = {1,1,1};
+
+
+        vec3 cam_origin = {0,0,0};
+        float port_width = 8;
+        float port_height = 8;
+        float focal_length = 2;
+        int max_col = 255;
+        Camera cam;
+
+
```

Comment:
Why are there so many blank lines here? 1 or 2 is fine, but this seems a bit excessive.